pypesto/objective/amici/amici.py

```diff
@@ -178,8 +178,12 @@ def __init__(
             parameter_mapping = create_identity_parameter_mapping(
                 amici_model, len(edatas)
             )
+        # parameter mapping where IDs of the currently fixed parameters
+        #  have been replaced by their respective values
+        #  (relevant for setting ``plist`` in ExpData later on)
         self.parameter_mapping = parameter_mapping
-
+        # parameter mapping independent of fixed parameters
+        self._parameter_mapping_full = copy.deepcopy(parameter_mapping)
```

Comment:
Curious, but is that always the full one? What if I now start by creating the amici objective with fixed parameters and then unfix a parameter from the problem?

Reply:
> What if I now start by creating the amici objective with fixed parameters and then unfix a parameter from the problem?

How would you do that? In terms of `Problem.fix_parameters`, I am not sure how you could start with fixed parameters before having had an objective without fixed parameters.

> Curious, but is that always the full one?

I guess, that depends on the definition of "full". Currently, it's "full" in the sense of "all parameters that were considered to be estimated at the time the AmiciObjective was created". Ever trying to change that to something "fuller" afterwards is probably not a good idea.

Reply:
mhh you are right i guess. I just checked the petabImporter. Was concerned how the mappings is done if petab has fixed parameters.